src/Symfony/Component/Security/Http/RememberMe/RememberMeHandlerInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\RememberMe;
+
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Handles creating and validating remember me cookies.
+ *
+ * If you want to add a custom implementation, you want to extend from
+ * {@see AbstractRememberMeHandler} instead.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @experimental in 5.3
+ */
+interface RememberMeHandlerInterface
+{
+    /**
+     * Creates a remember me cookie.
+     *
+     * The actual cookie should be set as an attribute on the master request,
+     * which is transformed into a response cookie by {@see ResponseListener}.
+     */
+    public function createRememberMeCookie(UserInterface $user): void;
```

Comment:
why not returning a `RememberMeDetails` to be leveraged for `$request->attributes->set(ResponseListener::COOKIE_ATTR_NAME, `

IIUC each handler needs to anticipate this now?

Reply:
I prefer this method to be fully self-contained, as it'll be used outside the Symfony framework (e.g. it's used by the SchebTwoFactorBundle to create the correct cookie after completing the 2fa).

Reply:
> it'll be used outside the Symfony framework

isnt that even more reason to solely expose the HTTP cookie here?